File: src/backend/Services/PasswordValidator.cs
using System.Text.RegularExpressions;

namespace ReactCore.Backend.Services;

public interface IPasswordValidator
{
    bool ValidatePassword(string password);
    string GetStrengthFeedback(string password);
}

public class PasswordValidator : IPasswordValidator
{
    public bool ValidatePassword(string password)
    {
        if (string.IsNullOrWhiteSpace(password) || password.Length < 8)
            return false;

        // Check for mixed case
        if (!Regex.IsMatch(password, "[a-z]") || !Regex.IsMatch(password, "[A-Z]"))
            return false;

        // Check for number
        if (!Regex.IsMatch(password, "[0-9]"))
            return false;

        // Check for special character
        if (!Regex.IsMatch(password, "[^a-zA-Z0-9]"))
            return false;

        return true;
    }

    public string GetStrengthFeedback(string password)
    {
        if (string.IsNullOrWhiteSpace(password))
            return "Password is required.";

        if (password.Length < 8)
            return "Password must be at least 8 characters long.";

        if (!Regex.IsMatch(password, "[a-z]") || !Regex.IsMatch(password, "[A-Z]"))
            return "Password must contain both uppercase and lowercase letters.";

        if (!Regex.IsMatch(password, "[0-9]"))
            return "Password must contain at least one number.";

        if (!Regex.IsMatch(password, "[^a-zA-Z0-9]"))
            return "Password must contain at least one special character.";

        return "Strong";
    }
}

File: src/frontend/src/utils/passwordValidator.ts
export interface PasswordValidationResult {
  isValid: boolean;
  errors: string[];
}

export const validatePassword = (password: string): PasswordValidationResult => {
  const errors: string[] = [];

  if (!password) {
    return { isValid: false, errors: ["Password is required."] };
  }

  if (password.length < 8) {
    errors.push("Password must be at least 8 characters long.");
  }
  if (!/[A-Z]/.test(password)) {
    errors.push("Password must contain at least one uppercase letter.");
  }
  if (!/[a-z]/.test(password)) {
    errors.push("Password must contain at least one lowercase letter.");
  }
  if (!/[0-9]/.test(password)) {
    errors.push("Password must contain at least one number.");
  }
  if (!/[^a-zA-Z0-9]/.test(password)) {
    errors.push("Password must contain at least one special character.");
  }

  return {
    isValid: errors.length === 0,
    errors
  };
};
